Welolo/client/src/index.js

```diff
@@ -5,32 +5,124 @@ import SuccessfulPayment from './SuccessfulPayment.js';
 
 
 class PaymentForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      fields: {},
+      errors: {},
+    };
+  }
+
+  handleValidation() {
+    let fields = this.state.fields;
+    let errors = {};
+    let formIsValid = true;
+
+    //recipient name
+    if (!fields["recipientname"]) {
+      formIsValid = false;
+      errors["recipientname"] = "Please insert the recipient name";
+    }
+
+    if (typeof fields["recipientname"] !== "undefined") {
+      if (!fields["recipientname"].match(/^[a-zA-Z]+$/)) {
+        formIsValid = false;
+        errors["recipientname"] = "Only letters are allowed";
+      }
+    }
+
+    if (fields["recipientname"].length > 30) {
+      formIsValid = false;
+        errors["recipientname"] = "Sorry, the name is too long";
+    }
+
+    //recipient phone
+    if (!fields["recipientph"]) {
+      formIsValid = false;
+      errors["recipientph"] = "Please insert the recipient phone number";
+    }
+
+    if (typeof fields["recipientph"] !== "undefined") {
+      if (!fields["recipientph"].match(/^\d+$/)) {
+        formIsValid = false;
+        errors["recipientph"] = "Only numbers are allowed";
+      }
+    }
+
+    if (fields["recipientph"].length > 10) {
+      formIsValid = false;
+        errors["recipientph"] = "Sorry, the phone number is too long";
+    }
+
+    //fund quantity
+    if (!fields["fundquantity"]) {
+      formIsValid = false;
+      errors["fundquantity"] = "Please insert the fund amount";
+    }
+
+    if (typeof fields["recipientph"] !== "undefined") {
+      if (!fields["recipientph"].match(/^\d+$/)) {
+        formIsValid = false;
+        errors["recipientph"] = "Only numbers are allowed";
+      }
+    }
+
+    //message
+    if (!fields["message"]) {
+      formIsValid = false;
+      errors["message"] = "Please insert a message";
+    }
+
+    this.setState({ errors: errors });
+    return formIsValid;
+  }
+
+  submitHandler(e) {
+    e.preventDefault();
+
+    if (this.handleValidation()) {
+      const data = new FormData(e.target);
+    
+      fetch('/', {
+        method: 'POST',
+        body: data,
+      });
+    } else {
+      alert("Please fill out all the inputs. Enter numbers for phone number and fund and letters for name and message.")
```

Comment:
It works great! The one thing I would recommend might be to print out the specific error messages to the screen or to the alert so the user knows what the issue is. It might be more work, but I liked the default "Please enter in a number" error you get when you enter in e into a numbers only text box. But that might be fancier than we are wanting to go.